packages/_private/vue-demo/src/DemoForm.vue

```diff
@@ -0,0 +1,323 @@
+  SynVueSelect,
+  SynVueSwitch,
+} from '@synergy-design-system/vue';
+import DemoFieldset from './DemoFieldset.vue';
+import { normalizeData } from './shared';
+
+type FormEnabledElements = HTMLElement & {
+  checked?: boolean;
+  name: string;
+  value?: string;
+};
+
+const initialFormData = {
+  code: '',
+  date: '',
+  email: '',
+  gender: '',
+  name: '',
+  newsletterAngular: false,
+  newsletterBeta: false,
+  newsletterReact: false,
+  newsletterStandard: false,
+  newsletterVanilla: false,
+  newsletterVue: true,
+  password: 'abAB1234',
+  phone: '',
+  role: '',
+  topics: [],
+};
+
+const formData = ref({
+  ...initialFormData,
+});
+
+const formRef = ref<HTMLFormElement>();
+
+const reset = () => {
+  console.log('here')
+  console.log(formData.value)
+  formData.value = {
+    ...initialFormData,
+  };
+  console.log(formData.value)
+}
+
+const submit = (e: Event) => {
+  e.preventDefault();
+  e.stopPropagation();
+
+  const formElement = e.target as HTMLFormElement;
+  const isValid = formElement.checkValidity();
+
+  if (isValid) {
+    // eslint-disable-next-line no-alert
+    alert('Your data was successfully submitted');
+  }
+}
+
+const synChange = (e: SynChangeEvent) => {
+  const form = formRef.value;
+
+  const normalizedData = normalizeData(new FormData(form));
+
+  // Log the normalized data
+  console.log(normalizedData);
+
+  // Set the field into state
+  const element = e.target as FormEnabledElements;
+  const { checked, name, value } = element;
+
+  const finalValue = typeof checked !== 'undefined'
+    ? !!checked
+    : value;
+
+  formData.value = {
+    ...formData.value,
+    [name]: finalValue,
+  }
+};
+</script>
+
+<template>
+  <form
+    @reset="reset"
+    @submit="submit"
+    @syn-change="synChange"
+    ref="formRef"
+  >
+    <!-- PersonalInformation -->
+    <DemoFieldset legend="Personal Information">
+
+      <SynVueRadioGroup
+        id="radiogroup-gender"
+        name="gender"
+        label="Please tell us your gender"
+        required
+        v-model="formData.gender"
+      >
+        <SynVueRadio value="f">Female</SynVueRadio>
+        <SynVueRadio value="m">Male</SynVueRadio>
+        <SynVueRadio value="other">Other</SynVueRadio>
+      </SynVueRadioGroup>
+
+      <SynVueSelect
+        id="select-role"
+        label="Current position"
+        name="role"
+        required
+        v-model="formData.role"
+      >
+        <SynVueOptgroup label="Developers">
+          <SynVueOption value="backend">Backend Developer</SynVueOption>
+          <SynVueOption value="frontend">Frontend Developer</SynVueOption>
+        </SynVueOptgroup>
+        <SynVueOptgroup label="Other">
+          <SynVueOption value="lead">Team Lead</SynVueOption>
+          <SynVueOption value="other">Other (please specify in comment section below)</SynVueOption>
+        </SynVueOptgroup>
+      </SynVueSelect>
+
+      <SynVueInput
+        id="input-text"
+        label="Name"
+        :minlength="10"
```

Comment:
5?

Reply:
I initially had problems getting this to work at all using TypeScript. Adjusted :)